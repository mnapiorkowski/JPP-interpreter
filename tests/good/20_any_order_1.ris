void main():
    let int a = 42;
    println add(a, y);

    first(0);

    return;

global int y = 2 * x;

int add(int a, int b):
    return a + b;

global int x = 10;

void first(int a):
    a++;
    print "first: ";
    println a;
    if (a < 10):
        second(a);
    endif;
    return; 

void second(int a):
    a++;
    print "second: ";
    println a;
    if (a < 10):
        first(a);
    endif;
    return; 