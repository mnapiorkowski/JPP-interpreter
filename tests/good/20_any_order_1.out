62
first: 1
second: 2
first: 3
second: 4
first: 5
second: 6
first: 7
second: 8
first: 9
second: 10
